sale_commission/models/sale_commission.py

```diff
@@ -36,6 +42,62 @@ class SaleCommission(models.Model):
     sections = fields.One2many(
         comodel_name="sale.commission.section", inverse_name="commission")
     active = fields.Boolean(default=True)
+    period = fields.Selection(
+        selection=[(PERIOD_MONTH, "Monthly"),
+                   (PERIOD_QUARTER, "Every quarter"),
+                   (PERIOD_SEMI, "Every semester"),
+                   (PERIOD_YEAR, "Every year")],
+        required=True, default=PERIOD_MONTH,
+    )
+    scope = fields.Selection(
+        selection=[("own_sales", "On his customer sales"),
+                   ("company_sales", "On company sales"),
+                   ("own_margin", "On his customers margin"),
+                   ("company_margin", "On company margin")],
+        required=True, default="own_sales",
+    )
+
+    def has_company_scope(self):
+        return self.scope in ("company_sales", "company_margin")
+
+    def has_own_scope(self):
+        return self.scope in ("own_sales", "own_margin")
+
+    agent_ids = fields.Many2many(
+        string="Agents", comodel_name="res.partner",
+        relation="agent_commission_rel",
+        column2="partner_id", column1="commission_id",
+    )
+
+    @api.multi
+    def compute_sale_commission(self, sale_line):
```

Comment:
This method doesn't make sense here, as it's refer to sales order line. Having computed fields very easy to handle on new API, I see no reason to use these helper methods.